File: client/iih_stub/iih.ci_stub/iih.ci.ord/ciorder/d/OrdNurseCareDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.ord.ciorder.d
{
    /// <summary>
    /// OrdNurseCareDO 的摘要说明。
    /// </summary>
    public class OrdNurseCareDO : BaseDO {

        public OrdNurseCareDO() {
        }
		public string Id_ornu {
            get { return getAttrVal<string>("Id_ornu",null); }
            set { setAttrVal<string>("Id_ornu", value); }
        }
		public string Id_or {
            get { return getAttrVal<string>("Id_or",null); }
            set { setAttrVal<string>("Id_or", value); }
        }
		public string Name3 {
            get { return getAttrVal<string>("Name3",null); }
            set { setAttrVal<string>("Name3", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_or_nu";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_ornu";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.ord.ciorder.d.OrdNurseCareDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_ornu", "Id_or", "Name3"};
        }
        
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/ciorder/d/OrdNurseCareDO.java
package iih.ci.ord.ciorder.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.ord.ciorder.d.desc.OrdNurseCareDODesc;
import java.math.BigDecimal;

/**
 * 临床医嘱护理 DO数据 
 * 
 */
public class OrdNurseCareDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_ORNU= "Id_ornu";
	public static final String ID_OR= "Id_or";
	public static final String NAME3= "Name3";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_ornu() {
		return ((String) getAttrVal("Id_ornu"));
	}	
	public void setId_ornu(String Id_ornu) {
		setAttrVal("Id_ornu", Id_ornu);
	}
	public String getId_or() {
		return ((String) getAttrVal("Id_or"));
	}	
	public void setId_or(String Id_or) {
		setAttrVal("Id_or", Id_or);
	}
	public String getName3() {
		return ((String) getAttrVal("Name3"));
	}	
	public void setName3(String Name3) {
		setAttrVal("Name3", Name3);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_ornu";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_or_nu";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(OrdNurseCareDODesc.class);
	}
	
}